tests/functional/test_package.py

```diff
@@ -804,11 +804,18 @@ def _write_fake_sdist(self, setup_py, directory, ext):
             with zipfile.ZipFile(path, 'w',
                                  compression=zipfile.ZIP_DEFLATED) as z:
                 z.writestr('sdist/setup.py', setup_py)
-        else:
+        elif ext == 'tar.gz':
             with tarfile.open(path, 'w:gz') as tar:
                 tarinfo = tarfile.TarInfo('sdist/setup.py')
                 tarinfo.size = len(setup_py)
                 tar.addfile(tarinfo, io.BytesIO(setup_py.encode()))
+        elif ext == 'tar.bz2':
+            with tarfile.open(path, 'w:bz2') as tar:
+                tarinfo = tarfile.TarInfo('sdist/setup.py')
```

Comment:
This is almost identical to the .tar.gz condition, can you consolidate the code here?